src/kafka.rs

```diff
@@ -13,22 +12,33 @@ enum KafkaError {
     InvalidPath { path: PathBuf },
 }
 
-#[derive(Clone, Copy, Debug, Derivative, Deserialize, Serialize)]
+/// Supported compression types for Kafka.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Derivative)]
 #[derivative(Default)]
 #[serde(rename_all = "lowercase")]
-pub(crate) enum KafkaCompression {
+pub enum KafkaCompression {
```

Comment:
Side note - I'm sorta surprised this isn't like the codecs where the component just opts for whichever it needs/supports. Feel like this could be a better shared piece of code.

Reply:
Kafka has its own set of supported compression schemes, so in this case, it really just needs an enum that can spit out a string value to feed to `librdkafka`.